apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -0,0 +1,134 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import SaveBar from '@cdo/apps/lib/levelbuilder/SaveBar';
+import {navigateToHref} from '@cdo/apps/utils';
+import $ from 'jquery';
+import color from '@cdo/apps/util/color';
+
+function useProgrammingExpression(initialProgrammingExpression) {
+  const [programmingExpression, setProgrammingExpression] = useState(
+    initialProgrammingExpression
+  );
```

Comment:
Whoa, cool! Excited to see these in action! https://reactjs.org/docs/hooks-overview.html#state-hook